Automattic/sensei
# Move private message form to contact teacher block

This adds the form rendering for the messaging the teacher inside the contact teacher block, so that it's displayed in a more predictable way. Everything else works the same as previously, with the exception of also being able to open the form via Javascript with a collapse/expand animation. The form is still submitted as a normal request.

### Changes proposed in this Pull Request

* Only display the Contact teacher block to logged in users, when messaging is not disabled in Sensei settings
* Add the `Send private message` form inside the contact teacher block, with minimal styling
* Display the form after the Contact teacher button/link when it's open
* Add collapse controls to the button to open/close the form
* Have non-JS fallback to display the form by clicking the link to `?contact=course`

### Testing instructions

* Add a Contact teacher block somewhere in the course, publish & view.
* Click contact teacher button/link
* Form should appear in place
* Type in message and send
* Confirmation notice should also appear in place, and the message should be sent


### Screenshot / Video

![QuickTime movie](https://user-images.githubusercontent.com/176949/96312095-38e2bb80-100b-11eb-97c3-9a1ffeddbc4c.gif)
